src/server/scripts/KulTiras/WaycrestManor/boss_lord_and_lady_waycrest.cpp

```diff
@@ -0,0 +1,554 @@
+    {
+        if (!UpdateVictim())
+            return;
+
+        events.Update(diff);
+
+        if (me->HasUnitState(UNIT_STATE_CASTING))
+            return;
+
+        while (uint32 eventId = events.ExecuteEvent())
+        {
+            ExecuteEvent(eventId);
+            if (me->HasUnitState(UNIT_STATE_CASTING))
+                return;
+        }
+    }
+
+private:
+    bool _isInArena;
+};
+
+// 261439 - Virulent Pathogen
+// 261441 - Virulent Pathogen (Expire)
+class spell_lord_and_lady_waycrest_virulent_pathogen_selector : public SpellScript
+{
+    bool Validate(SpellInfo const* /*spellInfo*/) override
+    {
+        return ValidateSpellInfo({ SPELL_VIRULENT_PATHOGEN_DAMAGE });
+    }
+
+    void HandleHitTarget(SpellEffIndex /*effIndex*/)
+    {
+        GetCaster()->CastSpell(GetHitUnit(), SPELL_VIRULENT_PATHOGEN_DAMAGE, true);
+    }
+
+    void Register() override
+    {
+        OnEffectHitTarget += SpellEffectFn(spell_lord_and_lady_waycrest_virulent_pathogen_selector::HandleHitTarget, EFFECT_0, SPELL_EFFECT_DUMMY);
+    }
+};
+
+// 261440 - Virulent Pathogen
+class spell_lord_and_lady_waycrest_virulent_pathogen_damage : public AuraScript
+{
+    bool Validate(SpellInfo const* /*spellInfo*/) override
+    {
+        return ValidateSpellInfo({ SPELL_VIRULENT_PATHOGEN_INFECT_AREA });
+    }
+
+    void HandleDamage(AuraEffect const* /*aurEff*/ , AuraEffectHandleModes /*mode*/)
+    {
+        if (GetTargetApplication()->GetRemoveMode() != AURA_REMOVE_BY_EXPIRE && GetTargetApplication()->GetRemoveMode() != AURA_REMOVE_BY_ENEMY_SPELL)
+            return;
+
+        Unit* target = GetTarget();
+        Unit* caster = GetCaster();
+        Map* map = target->GetMap();
+
+        target->CastSpell(target, SPELL_VIRULENT_PATHOGEN_INFECT_AREA, true);
+
+        if (map->IsHeroicOrHigher())
+            caster->CastSpell(target, SPELL_CONTAGIOUS_REMNANTS, false);
+    }
+
+    void Register() override
+    {
+        AfterEffectRemove += AuraEffectRemoveFn(spell_lord_and_lady_waycrest_virulent_pathogen_damage::HandleDamage, EFFECT_0, SPELL_AURA_PERIODIC_DAMAGE, AURA_EFFECT_HANDLE_REAL);
+    }
+};
+
+class DiscordantCadenzaDamageEvent : public BasicEvent
+{
+public:
+    explicit DiscordantCadenzaDamageEvent(Unit* caster, Position dest) : _caster(caster), _dest(dest) { }
+
+    bool Execute(uint64 /*time*/, uint32 /*diff*/) override
+    {
+        _caster->CastSpell(_dest, SPELL_DISCORDANT_CADENZA_DAMAGE, true);
+        return true;
+    }
+
+private:
+    Unit* _caster;
+    Position _dest;
+};
+
+// 268307 - Discordant Cadenza
+class spell_lord_and_lady_discordant_cadenza_selector : public SpellScript
+{
+    bool Validate(SpellInfo const* /*spellInfo*/) override
+    {
+        return ValidateSpellInfo({ SPELL_DISCORDANT_CADENZA_DAMAGE });
+    }
+
+    void HandleHitTarget(SpellEffIndex /*effIndex*/)
+    {
+        Unit* caster = GetCaster();
+        Unit* target = GetHitUnit();
+        float travelSpeed = 2;
```

Comment:
`static constexpr float TRAVEL_SPEED = 2000;`

you can use this in SendPlayOrphanSpellVisual (i assume you want 2s instead of 2ms (0.002s)

and use it as `Milliseconds(TRAVEL_SPEED)` in `AddEventAtOffset`